### ISSUE TITLE
Associated node geometry gets out of sync

### ISSUE THREAD

**Issue by alfarok** [alfarok/DynamoViewport#5]
When toggling previews on and off I have noticed geometry sometimes remains or gets linked with another node.  This leads me to believe the association between nodes and the container used to store each nodes geometry classified by the nodes GUID is getting out of sync.  Need further research into the issue.

**Comment by alfarok** [comment]
Here is at least 1 way to reproduce:

1) open delete.dyn (provided below)
2) update code block containing 65 to any other numerical value
3) toggle `Cube.ByLengths` visibility
4) toggle `Solid.Difference` visibility
5) toggle `Cube.ByLengths` visibility

The `Cube.ByLengths` will now display the `Solid.Difference` render package.  This likely means the GUID or groups are getting out of sync or an old/stale value is being send from the C# side.

**Possible areas to Investigate:**
[ViewportWindowViewModel.cs#L24](https://github.com/alfarok/DynamoViewport/blob/master/src/ViewportViewExtension/ViewportViewExtension/ViewModels/ViewportWindowViewModel.cs#L24)
or
[viewer.js#L219](https://github.com/alfarok/DynamoViewport/blob/master/src/ViewportViewExtension/ViewportViewExtension/Resources/assests/js/viewer.js#L219)

delete.dyn
```
{
  "Uuid": "0e728176-9856-4081-995b-500e2d2d01a0",
  "IsCustomNode": false,
  "Description": null,
  "Name": "delete",
  "ElementResolver": {
    "ResolutionMap": {
      "Cuboid": {
        "Key": "Autodesk.DesignScript.Geometry.Cuboid",
        "Value": "ProtoGeometry.dll"
      }
    }
  },
  "Inputs": [],
  "Outputs": [],
  "Nodes": [
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Sphere.ByCenterPointRadius@Autodesk.DesignScript.Geometry.Point,double",
      "Id": "a89c2e56d75b48dd8c07816f1e5cd350",
      "Inputs": [
        {
          "Id": "73ebf813061c44d99cd434304a1f0a46",
          "Name": "centerPoint",
          "Description": "Point\nDefault value : Autodesk.DesignScript.Geometry.Point.ByCoordinates(0, 0, 0)",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "77f4057197b44f939bcfdff3011426a1",
          "Name": "radius",
          "Description": "double\nDefault value : 1",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "3e4b6306270145699e36943d33f42ee1",
          "Name": "Sphere",
          "Description": "Sphere",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Create a Solid Sphere cetered at the input Point, with given radius.\n\nSphere.ByCenterPointRadius (centerPoint: Point = Autodesk.DesignScript.Geometry.Point.ByCoordinates(0, 0, 0), radius: double = 1): Sphere"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Cuboid.ByLengths@double,double,double",
      "Id": "6c4358f6e45e4c23bd3f47c4d8f178da",
      "Inputs": [
        {
          "Id": "81cad56e38c248ba9d1141e8791f2aaa",
          "Name": "width",
          "Description": "double\nDefault value : 1",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "b1cee5040fc543aa9891c394aae8c7dc",
          "Name": "length",
          "Description": "double\nDefault value : 1",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "27934949a0984efba3299bd9f1f37ea7",
          "Name": "height",
          "Description": "double\nDefault value : 1",
          "UsingDefaultValue": true,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "243d76beaf6f4d2494ccf2c43400a0d1",
          "Name": "Cuboid",
          "Description": "Cuboid",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "Create a Cuboid centered at WCS origin, with width, length, and height.\n\nCuboid.ByLengths (width: double = 1, length: double = 1, height: double = 1): Cuboid"
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.CodeBlockNodeModel, DynamoCore",
      "NodeType": "CodeBlockNode",
      "Code": "65;",
      "Id": "02f763f60ff14ef8a93f75e427140281",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "9297ddbd1a304de6a88bf220e74ab70d",
          "Name": "",
          "Description": "Value of expression at line 1",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "Allows for DesignScript code to be authored directly"
    },
    {
      "ConcreteType": "CoreNodeModels.Input.IntegerSlider, CoreNodeModels",
      "NodeType": "NumberInputNode",
      "NumberType": "Integer",
      "InputValue": 39,
      "MaximumValue": 100,
      "MinimumValue": 0,
      "StepValue": 1,
      "Id": "29ca6f93f89b43299b4a97e511fb4ee4",
      "Inputs": [],
      "Outputs": [
        {
          "Id": "7b2a53b3a38b44f2aa9cfe126c974789",
          "Name": "",
          "Description": "Int64",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Disabled",
      "Description": "A slider that produces integer values."
    },
    {
      "ConcreteType": "Dynamo.Graph.Nodes.ZeroTouch.DSFunction, DynamoCore",
      "NodeType": "FunctionNode",
      "FunctionSignature": "Autodesk.DesignScript.Geometry.Solid.Difference@Autodesk.DesignScript.Geometry.Solid",
      "Id": "8c4339106feb4730bfb7e3a0267f3ea9",
      "Inputs": [
        {
          "Id": "75b9902111204034b6fc7473e4096633",
          "Name": "solid",
          "Description": "Autodesk.DesignScript.Geometry.Solid",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        },
        {
          "Id": "9b9c6c555e2e4c4a9cbe8ea2d426cfcf",
          "Name": "other",
          "Description": "Solid",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Outputs": [
        {
          "Id": "e9a7a876d0254c69b08ffca4f36b7c63",
          "Name": "Solid",
          "Description": "Solid",
          "UsingDefaultValue": false,
          "Level": 2,
          "UseLevels": false,
          "KeepListStructure": false
        }
      ],
      "Replication": "Auto",
      "Description": "The boolean difference of this Solid with another\n\nSolid.Difference (other: Solid): Solid"
    }
  ],
  "Connectors": [
    {
      "Start": "3e4b6306270145699e36943d33f42ee1",
      "End": "9b9c6c555e2e4c4a9cbe8ea2d426cfcf",
      "Id": "e46a9ec3dd38441a9f1973fda1cb3d53"
    },
    {
      "Start": "243d76beaf6f4d2494ccf2c43400a0d1",
      "End": "75b9902111204034b6fc7473e4096633",
      "Id": "1f30ddf7dc124282a76662c300a6aff9"
    },
    {
      "Start": "9297ddbd1a304de6a88bf220e74ab70d",
      "End": "b1cee5040fc543aa9891c394aae8c7dc",
      "Id": "331c8bb2e39945748809aa25813926d3"
    },
    {
      "Start": "9297ddbd1a304de6a88bf220e74ab70d",
      "End": "81cad56e38c248ba9d1141e8791f2aaa",
      "Id": "963dc030bc6d431281a9a9d97a1fe720"
    },
    {
      "Start": "9297ddbd1a304de6a88bf220e74ab70d",
      "End": "27934949a0984efba3299bd9f1f37ea7",
      "Id": "bce3b0276ead4ae18cf37ad6373d7c21"
    },
    {
      "Start": "7b2a53b3a38b44f2aa9cfe126c974789",
      "End": "77f4057197b44f939bcfdff3011426a1",
      "Id": "7940c9d801714472bedd31a89b0b2664"
    }
  ],
  "Dependencies": [],
  "NodeLibraryDependencies": [],
  "Bindings": [],
  "View": {
    "Dynamo": {
      "ScaleFactor": 1.0,
      "HasRunWithoutCrash": true,
      "IsVisibleInDynamoLibrary": true,
      "Version": "2.9.0.2416",
      "RunType": "Automatic",
      "RunPeriod": "1000"
    },
    "Camera": {
      "Name": "Background Preview",
      "EyeX": -17.0,
      "EyeY": 24.0,
      "EyeZ": 50.0,
      "LookX": 12.0,
      "LookY": -13.0,
      "LookZ": -58.0,
      "UpX": 0.0,
      "UpY": 1.0,
      "UpZ": 0.0
    },
    "NodeViews": [
      {
        "Id": "a89c2e56d75b48dd8c07816f1e5cd350",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Sphere.ByCenterPointRadius",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 627.721923708091,
        "Y": 374.49731337281946
      },
      {
        "Id": "6c4358f6e45e4c23bd3f47c4d8f178da",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Cuboid.ByLengths",
        "ShowGeometry": true,
        "Excluded": false,
        "X": 665.0,
        "Y": 145.0
      },
      {
        "Id": "02f763f60ff14ef8a93f75e427140281",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Code Block",
        "ShowGeometry": true,
        "Excluded": false,
        "X": 391.0,
        "Y": 239.0
      },
      {
        "Id": "29ca6f93f89b43299b4a97e511fb4ee4",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Integer Slider",
        "ShowGeometry": true,
        "Excluded": false,
        "X": 250.72216090197912,
        "Y": 389.46487680946791
      },
      {
        "Id": "8c4339106feb4730bfb7e3a0267f3ea9",
        "IsSetAsInput": false,
        "IsSetAsOutput": false,
        "Name": "Solid.Difference",
        "ShowGeometry": false,
        "Excluded": false,
        "X": 923.98837374353161,
        "Y": 290.6166563270213
      }
    ],
    "Annotations": [],
    "X": -192.27921808855626,
    "Y": 103.39737041521551,
    "Zoom": 0.91048680047793251
  }
}
```

**Comment by alfarok** [comment]
Fixed for now with a band-aid: e68dd894f49796788eceda5ca77cdd67e811deae

Creating a more detailed follow-up task to address to deeper issue with synchronization.
